### PR TITLE
Refactor code for improved testability

### PR DESCRIPTION
**Pull request by jobu95**
Refactored Sessions so that IO is more cleanly separated from text processing, adding the process_response method to go between do_read and do_write. This should make the more sensitive part of the echo server - response preparation - more easily testable.

Also pulled nontrivial method definitions out of server.h and session.h, putting them into .cc files.

### CODE REVIEW

File: src/session.h [hunk 1]
```diff
@@ -43,61 +41,22 @@ class Session
...
-          }
-          });
-    }
+    /*
+     * Write msg to the client. On success, listen for more data.
+     */
+    void do_write(const char *msg, std::size_t length);
 
     tcp::socket socket_;
-    enum { max_length = 65536 }; // TODO: why is this an enum?
```
Comment (ahesselgrave) [inline src/session.h:99]: lol at the comment

File: src/server.cc [hunk 2]
```diff
@@ -4,3 +4,17 @@
 // file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
 #include "server.h"
+
+void Server::do_accept()
+{
+  acceptor_.async_accept(socket_,
+      [this](boost::system::error_code ec)
+      {
```
Comment (ahesselgrave) [inline src/server.cc:12]: Can we update the formatting to have tabs for scope? L12-19 is in the same tab region and for readability it should probably have some indentation.

File: src/session.cc [hunk 3]
```diff
@@ -4,3 +4,66 @@
...
+
+#include <math.h>
+
+void Session::do_read()
+{
+  auto self(shared_from_this());
+  socket_.async_read_some(boost::asio::buffer(data_, MAX_LENGTH),
+      [this, self](boost::system::error_code ec, std::size_t length)
+      {
+      if (!ec)
```
Comment (ahesselgrave) [inline src/session.cc:16]: Same indentation concern

File: src/session.cc [hunk 4]
```diff
@@ -4,3 +4,66 @@
...
+void Session::do_write(const char *msg, std::size_t length)
+{
+  auto self(shared_from_this());
+
+  debugf("do_write sending response of length %d\n", length);
+
+  boost::asio::async_write(socket_, boost::asio::buffer(msg,
+        length),
+      [this, self](boost::system::error_code ec, std::size_t /*length*/)
+      {
```
Comment (ahesselgrave) [inline src/session.cc:63]: One more tab indentation point

**Review by ahesselgrave — DISMISSED** [review]
Let's update the indentation to keep the codebase as clean and readable as possible. The logic is fine though

**Comment by ahesselgrave** [comment]
#10 builds on these changes. In the future, don't forget to close a pull request you are replacing with newer commits.

**Comment by ahesselgrave** [comment]
I'll leave the PR though in case I'm wrong. If that's the case, we can reopen.

File: src/server.cc (hunk 2 above)
Comment (jobu95) [reply src/server.cc:12]: Sure. I just used vim's built-in indentation engine. I'll keep an eye out for it now.

**Comment by jobu95** [comment]
Yeah my bad on that one, my git skills are pretty bad. I'll be more careful going forward.

**Pull request closed without merging**